### PR TITLE
Add nil check in bind and a test

### PR DESCRIPTION
**Pull request by lucasmrod**
This is the fix for #542. 

There are extensive tests in the existing `TestNilAndEmptyBytes`, but we cannot test the change in there because of how `sqlite3_column_blob()` works: _"The return value from sqlite3_column_blob() for a zero-length BLOB is a NULL pointer."_ -> this means that both `NULL` and `zero-length blob` map to a `nil` byte slice. So... I ended up adding a separate test `TestInsertNilByteSlice` to check that storing a nil byte slice is now storing a `NULL`.

### CODE REVIEW

File: sqlite3_test.go [hunk 1]
```diff
@@ -1580,6 +1580,21 @@ func TestNilAndEmptyBytes(t *testing.T) {
...
+		t.Fatal(err)
+	}
+	defer db.Close()
+	if _, err := db.Exec("create table blob_not_null (b blob not null)"); err != nil {
+		t.Fatal(err)
+	}
+	var nilSlice []byte
+	if _, err := db.Exec("insert into blob_not_null (b) values (?)", nilSlice); err == nil {
+		t.Fatal("didn't expect INSERT to 'not null' column with a nil []byte slice to work")
+	}
```
Comment (mattn) [inline sqlite3_test.go]: Could you please add test inserting nil too?

**Comment by mattn** [comment]
Thank you!

**Comment by GJRTimmer** [comment]
@lucasmrod Thanks, great work
